src/packet/signature/config.rs

```diff
@@ -2,20 +2,18 @@ use std::io::Read;
 
 use byteorder::{BigEndian, ByteOrder};
 use chrono::{DateTime, Utc};
-use derive_builder::Builder;
 use log::debug;
+use rand::{CryptoRng, Rng};
 
 use crate::crypto::hash::{HashAlgorithm, Hasher};
 use crate::crypto::public_key::PublicKeyAlgorithm;
-use crate::errors::{Error, Result};
+use crate::errors::Result;
 use crate::packet::{Signature, SignatureType, SignatureVersion, Subpacket, SubpacketData};
 use crate::ser::Serialize;
-use crate::types::{KeyId, PublicKeyTrait, SecretKeyTrait, Tag};
+use crate::types::{Fingerprint, KeyId, PublicKeyTrait, SecretKeyTrait, Tag};
 
-#[derive(Clone, PartialEq, Eq, Debug, Builder)]
-#[builder(build_fn(error = "Error"))]
+#[derive(Clone, PartialEq, Eq, Debug)]
 pub struct SignatureConfig {
-    #[builder(default)]
     pub version: SignatureVersion,
```

Comment:
I don't think we need this anymore, `version_specific` already encodes this

Reply:
Almost, with the exception of the distinction between V2 and V3.

But it might be nicer to add `V2` to the version specific enum, and drop `version` here.

Reply:
f06fc9d5617ff9190028e2327e39c640319892bb